### PR TITLE
UI/UX improvement

### PR DESCRIPTION
**Pull request by Kaktusryak**

### CODE REVIEW

**Comment by Kaktusryak** [comment]
1.  Added tab switches on swipes left/right
2.  Separated settings tab to user and settings pages available with more button on the tab
3.  On statistics page added feature when chart focuses on the category user select providing individual amount of category
4.  Created selector to get the amount of money is connected with selected category
6.  Restyle of settings/user pages items (created seperaded shareable item), visually separeted "dangerous" items
7.  At home page removed amount input, made "add" button on the center that invokes it's modal
8.  The add modal now looks according to figma with whole list of available categories presented on start
9.  The add modal appers with appropriate title (add/edit)
10.  Restyle of trash bin clear button on bin modal
11.  Restyle of Add category button on category management modal
12.  Binded update page data of home page to pull-n-drop of it's content (list of transactions)
13.  Removed png on background, added plain colour with gradiend (top-right)
14.  Removed statistics button on main page at top-right corner
15.  Changed Home tab button icon
16.  Sign pages look a bit better ( buttons are same size, inputs are equally wide with labels moving up while filling)
-----------------------------------
17.  Now 4 tabs
18.  Chart is working two way
19.  Fixed auth eyes
20.  Monobank settings are also on main page
21.  Styling of monbank settings 
------------------------------------
22.  Added google auth for both desktop and mobile (android)
23.  Moved details of spending to separated modal with all info about spend
24.  UI fixes

**Comment by Kaktusryak** [comment]
Now 4 tabs
Chart is working two way
Fixed auth eyes
Monobank settings are also on main page
Styling of monbank settings

File: src/app/auth/components/sign-in/sign-in.component.scss [hunk 1]
```diff
@@ -20,3 +21,24 @@ ion-input * {
...
 }
+
+.btns{
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  .btn{
+    width: 40%;
+  }
+}
```
Comment (YuriiKovalov) [inline src/app/auth/components/sign-in/sign-in.component.scss:32]: spasing

File: src/app/auth/components/sign-up/sign-up.component.scss [hunk 2]
```diff
@@ -3,7 +3,8 @@
 }
 
 ion-input {
-  --background: var(--page-background);
+  // --background: var(--page-background);
+  background-color: transparent !important;
```
Comment (YuriiKovalov) [inline src/app/auth/components/sign-up/sign-up.component.scss:7]: remove comment

File: src/app/core/state/selectors/spending.selectors.ts [hunk 3]
```diff
@@ -73,4 +73,11 @@ export namespace SpendingSelectors {
       return sortArrayByProperty(list, 'time', DirectionEnum.Descending);
     }
   );
+
+  export const selectStatCategoryAmount =(categoryId:string)=> createSelector(
+    selectStatSpendingList,
+    (list:SpendingModel[])=>{
+      return list.filter(el=>el.categoryId===categoryId).reduce((total,item)=>total+item.amount, 0) / 100;
+    } 
+  )
```
Comment (YuriiKovalov) [inline src/app/core/state/selectors/spending.selectors.ts:82]: spacing

File: src/app/pages/home-page/home.page.scss [hunk 4]
```diff
@@ -171,20 +172,20 @@ ion-content {
...
-  }
+  // .fixed:after {
+  //   content: '';
+  //   display: block;
+  //   top: -4px;
+  //   width: 100%;
+  //   height: 4px;
+  //   border-radius: 2px;
+  //   background-color: var(--background-color-3);
+  // }
```
Comment (YuriiKovalov) [inline src/app/pages/home-page/home.page.scss:184]: remove comment

File: src/app/pages/home-page/home.page.scss [hunk 5]
```diff
@@ -231,5 +232,10 @@ ion-content {
       max-height: 1px;
       overflow: hidden;
     }
+
+    .refresh{
+      width: calc(100% - 17px);
+      display: flex;
+    }
   }
```
Comment (YuriiKovalov) [inline src/app/pages/home-page/home.page.scss:240]: spacing

File: src/app/pages/home-page/home.page.ts [hunk 6]
```diff
@@ -25,13 +25,17 @@ import { SpendingService } from '../../core/services/spending/spending.service';
...
 
 @Component({
   selector: 'app-home-page',
   templateUrl: 'home.page.html',
   styleUrls: ['home.page.scss']
 })
 export class HomePage implements OnInit, OnDestroy {
+
+  @ViewChild('content', { static: false, read: ElementRef }) contentEl!: ElementRef;
+
```
Comment (YuriiKovalov) [inline src/app/pages/home-page/home.page.ts:38]: where contentEl is used?

File: src/app/core/services/spending/spending.service.ts [hunk 7]
```diff
@@ -106,6 +108,11 @@ export class SpendingService {
     this.store.dispatch(SpendingActions.updateHomePage({ payload: getCurrentMonthPeriodUNIX() }));
   }
 
+  updateHomePagePull(event: any): void {
+    this.store.dispatch(SpendingActions.updateHomePage({ payload: getCurrentMonthPeriodUNIX() }));
+    event.target.complete()
+  }
```
Comment (YuriiKovalov) [inline src/app/core/services/spending/spending.service.ts:114]: where this method is called?

File: src/app/pages/home-page/spending-basket-modal/spending-basket-modal.component.scss [hunk 8]
```diff
@@ -39,6 +39,10 @@ ion-content {
     }
   }
 
+  .delete-item{
+    --border-color:var(--ion-color-danger);
+  }
+
   .page-delimiter {
```
Comment (YuriiKovalov) [inline src/app/pages/home-page/spending-basket-modal/spending-basket-modal.component.scss:46]: spacing

File: src/app/pages/pages.component.ts [hunk 9]
```diff
@@ -1,60 +1,135 @@
...
+
 
 @Component({
   selector: 'app-pages',
   templateUrl: './pages.component.html',
   styleUrls: ['./pages.component.scss'],
 })
 export class PagesComponent implements OnDestroy {
+  @ViewChild('tabsRef', { static: false, read: ElementRef })
+  tabsRef!: ElementRef;
```
Comment (YuriiKovalov) [inline src/app/pages/pages.component.ts:32]: rewrite to one row

File: src/app/pages/pages.component.ts [hunk 10]
```diff
@@ -1,60 +1,135 @@
...
+        payload: {
+          startDate: moment().startOf(ViewPeriod.Month).unix(),
+          endDate: moment().endOf(ViewPeriod.Month).unix(),
+        },
+      })
+    );
+    this.subscription.add(
+      this.store
+        .select(UserSelectors.selectUser)
+        .pipe(take(2))
```
Comment (YuriiKovalov) [inline src/app/pages/pages.component.ts:77]: invistigate why here is take(2)

File: src/app/pages/pages.component.ts [hunk 11]
```diff
@@ -1,60 +1,135 @@
...
+        .subscribe(
+          (value) =>
+            value?.monoBankAccounts?.length &&
+            this.store.dispatch(BankAccountsActions.checkWebHook())
+        )
```
Comment (YuriiKovalov) [inline src/app/pages/pages.component.ts:82]: rewrite to one row

File: src/app/pages/settings-page/settings.page.scss [hunk 12]
```diff
@@ -4,3 +4,11 @@ ion-header {
...
   }
 }
+.settings-list{
+  padding: 0;
+  min-height: fit-content;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+}
```
Comment (YuriiKovalov) [inline src/app/pages/settings-page/settings.page.scss:14]: spacing

File: src/app/pages/statistics-page/statistics.page.ts [hunk 13]
```diff
@@ -33,56 +46,111 @@ export class StatisticsPage implements OnInit, OnDestroy {
...
-    public spendingService : SpendingService,
+    public spendingService: SpendingService
   ) {
     this.formGroup = this.fb.group({
       periodRange: this.fb.control(null),
-      monthControl: this.fb.control(this.availableMonthsInCurrentYear.slice(-1)[0]),
-      yearControl: this.fb.control(this.availableYears.slice(-1)[0])
+      monthControl: this.fb.control(
+        this.availableMonthsInCurrentYear.slice(-1)[0]
+      ),
```
Comment (YuriiKovalov) [inline src/app/pages/statistics-page/statistics.page.ts:96]: to one row

File: src/app/pages/statistics-page/statistics.page.ts [hunk 14]
```diff
@@ -33,56 +46,111 @@ export class StatisticsPage implements OnInit, OnDestroy {
...
-        this.updateList();
-      })
+          this.store.dispatch(SpendingActions.updateStatTimePeriod({
+              payload: { startDate, endDate },
+            })
+          );
+          this.updateList();
+        }
+      )
     );
```
Comment (YuriiKovalov) [inline src/app/pages/statistics-page/statistics.page.ts:154]: a lot of logic in OnInit, make sense split these on several methods and call them in oninit (functional approuch)

File: src/app/pages/statistics-page/statistics.page.ts [hunk 15]
```diff
@@ -95,19 +163,82 @@ export class StatisticsPage implements OnInit, OnDestroy {
...
+        if(this.selectedCategoryIndex != null || this.selectedCategoryIndex == index){
+          this.selectedCategoryIndex = null
+          this.categoryAmount$ = null;
+        }else{
+          this.selectedCategoryIndex = index
+          this.subscription.add(this.spendingByCategoriesList$.subscribe(categories=>{
+            this.categoryAmount$ = of(categories[index].totalAmount / 100)
+          })) 
+        }
+      }else{
```
Comment (YuriiKovalov) [inline src/app/pages/statistics-page/statistics.page.ts:221]: spacing

File: src/app/pages/user-page/user.page.html [hunk 16]
```diff
@@ -0,0 +1,24 @@
+<ion-header>
+  <ion-toolbar>
+    <ion-title>{{ 'user.title' | translate}}</ion-title>
+  </ion-toolbar>
```
Comment (YuriiKovalov) [inline src/app/pages/user-page/user.page.html:4]: spacing

File: src/app/pages/user-page/user.page.scss [hunk 17]
```diff
@@ -0,0 +1,15 @@
+ion-header {
+
+  ion-title.md {
+    padding-left: 40px;
+  }
+}
+
+.settings-list{
+  padding: 0;
```
Comment (YuriiKovalov) [inline src/app/pages/user-page/user.page.scss:9]: spacing

File: src/app/shared/components/configure-spending-modal/configure-spending-modal.component.html [hunk 18]
```diff
@@ -3,7 +3,10 @@
     <ion-buttons slot="start">
       <ion-button color="danger" (click)="cancel()">{{ 'general.actions.cancel' | translate }}</ion-button>
     </ion-buttons>
-    <ion-title>{{ 'editSpendingModal.title' | translate }}</ion-title>
+    <ion-title *ngIf="type!==actionsEnum.Add;else add">{{ 'editSpendingModal.titleEdit' | translate }}</ion-title>
```
Comment (YuriiKovalov) [inline src/app/shared/components/configure-spending-modal/configure-spending-modal.component.html:6]: spacing

File: src/app/shared/components/configure-spending-modal/configure-spending-modal.component.scss [hunk 19]
```diff
@@ -23,3 +29,49 @@ ion-input {
...
+  color: var(--button-content-color-1);
+}
+.categories{
+  ion-item{
+    --border-style:none;
+  }
+  .picked{
+    --background-color: red;
+  }
+}
```
Comment (YuriiKovalov) [inline src/app/shared/components/configure-spending-modal/configure-spending-modal.component.scss:42]: spacing

File: src/app/shared/components/configure-spending-modal/configure-spending-modal.component.ts [hunk 20]
```diff
@@ -40,6 +43,13 @@ export class ConfigureSpendingModalComponent implements OnInit {
       category: this.fb.control(this.spendingItem?.category || null, Validators.required),
       description: this.fb.control(this.spendingItem?.description || null),
     });
+    this.selectedCategory = this.spendingItem?.category || null
+  }
+
+  ionViewDidEnter() {
+    setTimeout(() => {
+      this.inputElement.nativeElement.focus();
+    }, 200);
```
Comment (YuriiKovalov) [inline src/app/shared/components/configure-spending-modal/configure-spending-modal.component.ts:52]: Do you sure is the timeout is the best way?

**Pull request merged**